novawallet/Modules/Notifications/GovernanceNotifications/GovernanceNotificationsPresenter.swift

```diff
@@ -0,0 +1,126 @@
+import Foundation
+import RobinHood
+import SoraFoundation
+
+final class GovernanceNotificationsPresenter {
+    weak var view: GovernanceNotificationsViewProtocol?
+    let wireframe: GovernanceNotificationsWireframeProtocol
+    let interactor: GovernanceNotificationsInteractorInputProtocol
+    private let chainList: ListDifferenceCalculator<ChainModel>
+    private var settings: [ChainModel.Id: GovernanceNotificationsModel]
+
+    init(
+        settings: [ChainModel.Id: GovernanceNotificationsModel],
+        interactor: GovernanceNotificationsInteractorInputProtocol,
+        wireframe: GovernanceNotificationsWireframeProtocol
+    ) {
+        self.settings = settings
+        self.interactor = interactor
+        self.wireframe = wireframe
+
+        chainList = ListDifferenceCalculator(initialItems: []) { chain1, chain2 in
+            ChainModelCompator.defaultComparator(chain1: chain1, chain2: chain2)
+        }
+    }
+
+    private func provideViewModels() {
+        let viewModel = chainList.allItems.map {
+            if let chainSettings = settings[$0.identifier] {
+                return GovernanceNotificationsModel(
+                    identifier: $0.identifier,
+                    enabled: chainSettings.enabled,
+                    icon: RemoteImageViewModel(url: $0.icon),
+                    name: $0.name,
+                    newReferendum: chainSettings.newReferendum,
+                    referendumUpdate: chainSettings.referendumUpdate,
+                    delegateHasVoted: chainSettings.delegateHasVoted,
+                    tracks: chainSettings.tracks
+                )
+            } else {
+                let newModel = GovernanceNotificationsModel(
+                    identifier: $0.identifier,
+                    enabled: false,
+                    icon: RemoteImageViewModel(url: $0.icon),
+                    name: $0.name,
+                    newReferendum: true,
+                    referendumUpdate: true,
+                    delegateHasVoted: true,
+                    tracks: .all
+                )
+                settings[$0.identifier] = newModel
```

Comment:
I think it is better populate settings in the separate function and don't break functional paradigm with side effects